File: certificates/problem-solving-basic/balanced-system-files-partition/mostBalancedPartition.js
function mostBalancedPartition(parent, files_size) {
  for (let i = parent.length - 1; i > 0; i--) {
    files_size[parent[i]] += files_size[i];
  }

  let mindiff = files_size[0];
  for (let i = 1; i < parent.length; i++) {
    let diff = Math.abs(files_size[0] - 2 * files_size[i]);
    if (diff < mindiff) {
      mindiff = diff;
    }
  }

  return mindiff;
}

File: certificates/problem-solving-basic/balanced-system-files-partition/mostBalancedPartition.c
#include <stdio.h>
#include <stdlib.h>

int mostBalancedPartition(int parent_count, int* parent, int files_size_count, int* files_size)
{

  for (int i = parent_count - 1; i > 0; i--)
  {
    files_size[parent[i]] += files_size[i];
  }

  int mindiff = files_size[0];
  int diff = mindiff;
  for (int i = 1; i < files_size_count; i++)
  {
    diff = abs(files_size[0] - 2 * files_size[i]);
    mindiff = (diff < mindiff) ? diff : mindiff;
  }
  return mindiff;
}